bigquery_etl/cli/stage.py

```diff
@@ -169,50 +176,48 @@ def deploy(
             artifact_file.rename(query_path)
             artifact_file = query_path
 
-        new_artifact_path = Path(sql_dir) / project_id / dataset / name
+        new_artifact_path = (
+            Path(sql_dir) / project_id / new_artifact_dataset / artifact_name
+        )
         new_artifact_path.mkdir(parents=True, exist_ok=True)
         shutil.copytree(artifact_file.parent, new_artifact_path, dirs_exist_ok=True)
         updated_artifact_files.add(new_artifact_path / artifact_file.name)
 
         # copy tests to the right structure
-        if test_path.exists():
-            test_destination = TEST_DIR / project_id / dataset / name
-            shutil.copytree(test_path, test_destination, dirs_exist_ok=True)
+        if artifact_test_path.exists():
+            new_artifact_test_path = (
+                TEST_DIR / project_id / new_artifact_dataset / artifact_name
+            )
+            shutil.copytree(
+                artifact_test_path, new_artifact_test_path, dirs_exist_ok=True
+            )
             if remove_updated_artifacts:
-                shutil.rmtree(test_path)
-
-            # rename test files
-            for test_file_path in map(Path, glob(f"{TEST_DIR}/**/*", recursive=True)):
-                for test_dep_file in artifact_files:
-                    test_project = test_dep_file.parent.parent.parent.name
-                    test_dataset = test_dep_file.parent.parent.name
-                    test_name = test_dep_file.parent.name
-
-                    file_suffix = test_file_path.suffix
-                    if test_file_path.name in (
-                        f"{test_project}.{test_dataset}.{test_name}{file_suffix}",
-                        f"{test_project}.{test_dataset}.{test_name}.schema{file_suffix}",
-                    ) or (
-                        test_file_path.name
-                        in (
-                            f"{test_dataset}.{test_name}{file_suffix}",
-                            f"{test_dataset}.{test_name}.schema{file_suffix}",
-                        )
-                        and test_project == project
-                    ):
-                        test_dataset = f"{test_dataset}_{project.replace('-', '_')}"
-
-                        if dataset_suffix:
-                            test_dataset = f"{test_dataset}_{dataset_suffix}"
-
-                        name = f"{project_id}.{test_dataset}.{test_name}"
-                        if test_file_path.name.endswith(f".schema{file_suffix}"):
-                            name += ".schema"
-                        name += file_suffix
+                shutil.rmtree(artifact_test_path)
+
+        # rename test files
+        for test_file_path in map(Path, glob(f"{TEST_DIR}/**/*", recursive=True)):
```

Comment:
Re-scanning all test files for every deployed artifact isn't ideal, so I'm going to refactor this to process the test files in a single pass after the main artifact processing loop.

One minor downside is the staging dataset suffix logic will be duplicated, but that was already the case before with the existing logic, so I don't see that as a dealbreaker.